utils.py

```diff
@@ -1,40 +1,41 @@
 import signal
 import sys
-import datetime
+import typing
+from typing import Optional, Union
 
-from sqlalchemy import create_engine
+from sqlalchemy import create_engine, orm
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.orm import sessionmaker
 
 from core.base import DATABASE_CONNECTION_PATH
+from datetime import datetime
 
 
-def get_database_connection(database_connection_path=DATABASE_CONNECTION_PATH):
+def get_database_connection(database_connection_path: str=DATABASE_CONNECTION_PATH) -> orm.session.Session:
     """ Unlike database file paths, database connection paths must begin with
     sqlite:/// """
     assert database_connection_path.startswith('sqlite:///'), 'Database connection path must start with sqlite:///'
 
     engine = create_engine(database_connection_path)
-    Base = declarative_base()
-    Base.metadata.bind = engine
     Session = sessionmaker(bind=engine)
 
     return Session()
 
-
-def remove_empty_from_dict(d):
+def remove_empty_from_dict(d: Union[dict, list, typing.Any]) -> Union[dict, list, typing.Any]:
     """ Reference: https://stackoverflow.com/a/24893252 """
     if type(d) is dict:
+        d = typing.cast(dict, d)
```

Comment:
This is mostly for mypy to know it should be a dict. Don't believe it does anything at runtime.